test/test_profiler.py

```diff
@@ -1282,6 +1288,54 @@ def test_utils_compute_queue_depth_when_no_cuda_events(self):
         basic_evaluation = _utils.BasicEvaluation(prof.profiler)
         self.assertFalse(basic_evaluation.compute_queue_depth())
 
+    def test_utils_compute_idle_time(self):
+        profiler = self.generate_mock_profile()
+        basic_evaluation = _utils.BasicEvaluation(profiler)
+        res = ""
+        for event_key in basic_evaluation.event_keys:
+            res += f"{basic_evaluation.metrics[event_key].idle_time_ns} [{event_key.event.name()}]\n"
+
+        self.assertExpectedInline(
+            res, """\
+0 [CPU (Before cudaLaunchKernel)]
+0 [CPU (Before cudaLaunchKernel)]
+0 [CPU (Before cudaLaunchKernel)]
+0 [CPU (Before cudaLaunchKernel)]
+0 [CPU (After cudaLaunchKernel)]
+0 [CPU (After cudaLaunchKernel)]
+0 [CPU (After cudaLaunchKernel)]
+0 [CPU (After cudaLaunchKernel)]
+0 [CPU (After GPU)]
+0 [CPU (After GPU)]
+100000 [CPU (After GPU)]
+300000 [CPU (No Event)]
+""")
+
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is required")
+    def test_utils_get_optimizable_events(self):
```

Comment:
I'm concerned about this test for two reasons:
1) It's vulnerable to flakiness the same way the previous flaky one was
2) It's an E2E test, so it won't catch small changes until something puts it over the edge.

One thing that you might consider doing is running the evaluation just to make sure it runs and maintains the stable invariants (like `len(optimizable_events) == 5`), and then use the results to generate a mock artifact for detailed deterministic testing. If you gate the check on `expecttest.ACCEPT` it will only be used to update, and you can use the `skip` argument to store the literal in a variable.

Reply:
I am also not certained on the test for this one:
1. Heuristic changes (cost function change) require to write a  new set of mock test
2. len(optimizable_events) == 5 cannot be guaranteed because we are filtering the events outside downward interval. If due to flakiness we filter out all the event, the invariant is gone too. :(

Reply:
1) If we change the cost function won't we just have to update the golden values? (I'm glad you're thinking about that.)
2) Ah, right.